diff --git a/dist/components/charts/PieChart/PieChart.base.svelte b/dist/components/charts/PieChart/PieChart.base.svelte
index 62198a1924b1d00471c72c90d979d2dcc6247e4f..6ab55af4e475efe3f2c671d7dc76891292b2a43b 100644
--- a/dist/components/charts/PieChart/PieChart.base.svelte
+++ b/dist/components/charts/PieChart/PieChart.base.svelte
@@ -56,7 +56,7 @@
     props = {},
     profile = false,
     tooltipContext = true,
-    marks,
+    marks: marksProp,
     tooltip: tooltipProp,
     pie,
     arc,
@@ -223,8 +223,8 @@
   }}
 >
   {#snippet marks(snippetProps: any)}
-    {#if typeof marks === 'function'}
-      {@render marks(snippetProps)}
+    {#if typeof marksProp === 'function'}
+      {@render marksProp(snippetProps)}
     {:else}
       <Group {...getGroupProps()}>
         <!-- Use `series` instead of `visibleSeries` since data is filtered (legend) instead of series -->
